File: C++ Codes/InsertionSort.cpp
#include <iostream>
using namespace std;

void InsertionSort(int a[],int n)
{
   int i, key, j;
    for (i = 1; i < n; i++)
    {
        key = a[i];
        j = i - 1;
 
       
        while (j >= 0 && a[j] > key)
        {
            a[j + 1] = a[j];
            j = j - 1;
        }
        a[j + 1] = key;
    }

}

int main()
{
    int n;
    cout<<"Enter size of list::";
    cin>>n;
    int arr[n];

    cout<<"Enter the list::";
    for(int j=0;j<n;j++)
    {
        cin>>arr[j];
    }
    InsertionSort(arr,n);
  for(int k=0;k<n;k++)
  {
      cout<<arr[k]<<" ";
  }

  return 0;
}

File: Python Codes/WEEK 9 Sorting Algorithms/insertion_sort.py
#Insertion Sorting Implementation

def insertion_sort(arr):
    
    for i in range(1,len(arr)):

        key = arr[i]

        j=i-1

        while j>=0 and key<arr[j]:
            arr[j+1] = arr[j]
            j-=1
        
        arr[j+1] = key

arr=[8,3,10,2,5,0]

insertion_sort(arr)

print(arr)